featurebyte/storage/base.py

```diff
@@ -67,22 +72,58 @@ async def try_delete_if_exists(self, remote_path: Path) -> None:
         except FileNotFoundError:
             pass
 
-    @abstractmethod
-    async def get(self, remote_path: Path, local_path: Path) -> None:
+    async def get(self, remote_path: Path, local_path: Path, cache_key: Any = None) -> None:
         """
-        Download file from storage to local path
+        Download file from storage to local path with caching if cache_key is provided
 
         Parameters
         ----------
         remote_path: Path
             Path of remote file to be downloaded
         local_path: Path
             Path to stored downloaded file
+        cache_key: Any
+            Cache key for storing downloaded file (if provided, the result will be cached).
+            Please do not use cache_key for large files.
```

Comment:
Typically we want to use caching for files that can be large such as feature clusters and registry, is that correct?

Reply:
Since we want to load the file into memory, we don't want to load like 1GB data into memory which can cause other issue. Feature clusters & registry is more like in several 10MB range, which is much smaller than 1GB.

Reply:
What do you think about caching the file on disk instead? Then the lru cache can track the cached filename and doesn't need to keep file content in memory.

Reply:
In this case, maybe we put it under `.featurebyte/cache` directory?

Reply:
I think i have some ideas, let's me revise this & ping you once it is revised.

Reply:
Either seems fine to me. I think we should check whether the file still exists upon cache hit.